content/src/ports/activeEntities.ts

```diff
@@ -0,0 +1,175 @@
+import { Deployment, Entity, EntityId, Pointer } from 'dcl-catalyst-commons'
+import LRU from 'lru-cache'
+import { EnvironmentConfig } from '../Environment'
+import { getDeployments } from '../service/deployments/deployments'
+import { AppComponents } from '../types'
+
+export type ActiveEntities = {
+  withPointers(...pointers: Pointer[]): Promise<Entity[]>
+  withIds(...entityIds: EntityId[]): Promise<Entity[]>
+  inactivate(...pointers: Pointer[]): void
+  activate(pointer: Pointer, entity: Entity): void
+}
+
+const mapDeploymentsToEntities = (deployments: Deployment[]): Entity[] => {
+  return deployments.map(({ entityVersion, entityId, entityType, pointers, entityTimestamp, content, metadata }) => ({
+    version: entityVersion,
+    id: entityId,
+    type: entityType,
+    pointers,
+    timestamp: entityTimestamp,
+    content: content?.map(({ key, hash }) => ({ file: key, hash })),
+    metadata
+  }))
+}
+
+export const createActiveEntitiesComponent = (
+  components: Pick<AppComponents, 'database' | 'env' | 'logs' | 'metrics'>
+): ActiveEntities => {
+  const logger = components.logs.getLogger('ActiveEntities')
+  const cache = new LRU<EntityId, Entity>({
+    max: components.env.getConfig(EnvironmentConfig.ENTITIES_CACHE_SIZE)
+  })
+  const entityIdByPointers = new Map<Pointer, EntityId>()
```

Comment:
Will be a limit for each entity type here? It'd be a pity to store stale profiles or use that space that should be a scene.